### ISSUE TITLE
runtime assignment should throw an error instead of breaking the kerner

### ISSUE THREAD

**Issue by arcanist123** [abaplint/transpiler#427]
This code
```abap
        TYPES:
        BEGIN OF ts_str,
        comp_one TYPE i,
        END OF ts_str.
    DATA lv_int1 TYPE i VALUE 1.
    DATA lv_str TYPE ts_str.

    FIELD-SYMBOLS <ls_str> type any.
    ASSIGN lv_str to <ls_str>.
    MOVE lv_int1 TO <ls_str>.
    WRITE <ls_str>.
```
should throw a meaningful error instead of breaking internally

**Comment by arcanist123** [comment]
same as this one
``` abap 
  TYPES:
    BEGIN OF ts_str,
      comp_one TYPE i,
    END OF ts_str.
  DATA lv_string TYPE string.
  DATA lv_str TYPE ts_str.

  FIELD-SYMBOLS <ls_str> type any.
  ASSIGN lv_str to <ls_str>.
  MOVE lv_string TO <ls_str>.
  WRITE <ls_str>.
```

**Comment by arcanist123** [comment]
and this one
```abap
REPORT z_test_transpiler.

CLASS lcl_ DEFINITION CREATE PUBLIC.

  PUBLIC SECTION.
ENDCLASS.

CLASS lcl_ IMPLEMENTATION.
ENDCLASS.

START-OF-SELECTION.
  DATA lo_aaa TYPE REF TO lcl_.
  field-SYMBOLS <lo_aaa> type any.
  ASSIGN lo_aaa to <lo_aaa>.
  WRITE: <lo_aaa>.
```
